### PR TITLE
Model and migration for 1:n SCC-Katello-Repo association

### PR DESCRIPTION
**Pull request by nadjaheitmann**

### CODE REVIEW

File: db/migrate/20220425132605_add_scc_katello_repository_relation.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AddSccKatelloRepositoryRelation < ActiveRecord::Migration[6.0]
+  def change
+    create_table(:scc_katello_repositories) do |t|
+      t.integer :scc_repository_id, null: false
+      t.integer :katello_root_repository_id, null: false
```
Comment (m-bucher) [inline db/migrate/20220425132605_add_scc_katello_repository_relation.rb:5]: Is it possible to use `t.references`, here? See https://api.rubyonrails.org/v6.0/classes/ActiveRecord/ConnectionAdapters/TableDefinition.html#method-i-references

**Review by m-bucher — CHANGES_REQUESTED** [review]
Is there a way we can merge the DB-migrations into a single one?

File: app/lib/actions/scc_manager/subscribe_product.rb [hunk 2]
```diff
@@ -38,7 +38,7 @@ def finalize
         # extract Katello repo ids from input hash and store to database
         input[:katello_repos].each do |scc_repo_id, katello_root_repository_id|
           scc_repo = SccRepository.find(scc_repo_id)
-          scc_repo.update!(katello_root_repository_id: katello_root_repository_id)
+          scc_repo.update!(katello_root_repository_ids: scc_repo.katello_root_repository_ids.push(katello_root_repository_id))
```
[the remark this answers is not in the document]
Comment (nadjaheitmann) [reply app/lib/actions/scc_manager/subscribe_product.rb:41]: Not sure, if this works to be honest... I did not add a reference from root repositories to scc_repositories that could handle this in Rails

**Review by m-bucher — APPROVED** [review]
Squash the commits and it is good to go :smile:

**Pull request merged**